homeassistant/components/miraie_ac/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow for MirAIe AC integration."""
+from __future__ import annotations
+
+import logging
+import re
+from typing import Any
+
+from py_miraie_ac import (
+    AuthException,
+    AuthType,
+    ConnectionException,
+    MirAIeAPI,
+    MobileNotRegisteredException,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import CONFIG_KEY_PASSWORD, CONFIG_KEY_USER_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONFIG_KEY_USER_ID): str,
+        vol.Required(CONFIG_KEY_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    login_id = data[CONFIG_KEY_USER_ID]
+    password = data[CONFIG_KEY_PASSWORD]
+
+    async with MirAIeAPI(
+        auth_type=AuthType.MOBILE, login_id=login_id, password=password
+    ) as api:
+        try:
+            _LOGGER.debug("Initializing MirAIe API")
+
+            if not _validate_mobile_number(login_id):
+                raise ValidationError()
+
+            await api.initialize()
+        except ValidationError:
+            _LOGGER.error("Invalid Mobile Number")
+            raise
+        except AuthException:
+            _LOGGER.error("Invalid user ID or password")
+            raise
+        except ConnectionException:
+            _LOGGER.error("Error connecting to the MirAIe services")
+            raise
+        except MobileNotRegisteredException:
+            _LOGGER.error("Mobile number not registered with MirAIe")
+            raise
```

Comment:
As you reraise the exceptions, the logging can also be moved to the parent method, where you handle the exceptions.